Assets/Plugins/EuclideonUdSDK/udSDK/csAPI/udSDKAPI.cs

```diff
@@ -509,17 +535,42 @@ public RenderOptions() : this(udRenderContextPointMode.udRCPM_Rectangles, udRend
 
         public void setPick(uint x, uint y)
         {
-            pick = new udRenderPicking();
+            udRenderPicking pick = new udRenderPicking();
             pick.x = x;
             pick.y = y;
-            Marshal.StructureToPtr(pick, options.pPick, true);
+
+            Marshal.StructureToPtr(pick, options.pPick, false);
 
             pickRendered = false;
             pickSet = true;
+        }
 
+        public udPick getPick()
+        {
+            UnityEngine.Debug.Log("Getting pick");
+
+            if(!pickSet)
+                return new udPick();
+
+            udRenderPicking targetPick = this.Pick;
```

Comment:
Does `targetPick` now live in managed memory, or unmanaged memory? (I assume it is unmanaged, since we are copying to `udPick`)

Reply:
The original is unmanaged, the copy is in managed

Reply:
It is managed, because the block it's being declared in is safe - it should be getting cleaned up after getPick is done.
The only variable living in unmanaged memory connected to RenderOptions is options.pPick.